### PR TITLE
Grammar bugfixes, new features, and refactoring

### PR DESCRIPTION
**Pull request by falkecarlsen**
Bugfixes:
- Missing definitions of `PAREN_START` and `PAREN_END` resulting in implicit left-recursion - it did compile with warnings before this fix
- Fixed literal declaration of `SQ_BRACKET_START` in `board_world_dim`-definition
- Threw out dreamcode in `example.cell` that would not easily express the same notion without much progress on grammar
- Fixed spelling mistake. The proper abbreviation for parenthesis is `PAREN` - [supposedly](https://www.allacronyms.com/parenthesis/abbreviated).

New features:
- Added grammar-definition for single-line comments with `//` followed by optionally whatever that isn't carriage- or newline-meta
- Added grammar-definition for multi-line comments with `/*` optionally whatever `*/`
- Added support for `C`-like post- and pre-fix increment and decrement. Note that previous iteration supported post-fix increment/decrement, but only as a part of an expression.

I'm not certain that the implementation of post/pre inc- and decrement is very good.

Also renamed both grammar and filename to a non-test name, just `cellmata` for now, which allows the `antlr4` intellij-plugin to test the grammar and create parse-trees from programs written in this language-syntax. Here a parse-tree for `example2.cell`, which is wildly inflated at expressions, due to current grammar-structure:
image

### CODE REVIEW

**Review by NicEastvillage — APPROVED** [review]
I approve. But I think you should wait for @jenrik 's approval

**Comment by falkecarlsen** [comment]
Agreed. Hence the PR and not just commits on `master`. Would also like to hear @jenrik take on the structuring of the increment/decrement-statement.

File: cellumata_syntax-v2/cellmata.g4
Comment (jenrik) [inline cellumata_syntax-v2/cellmata.g4]: Missing support for incrementing array fields. For example `baz[0]--;` should be valid but isn't. We should probably create a new production that includes both `var_ident` and `array_lookup`.

**Review by jenrik — CHANGES_REQUESTED** [review]
The comment implementation seem good but we are allowing comments in some potentially problematic places. Lets the discuss the comment implementation on monday.
The increment and decrement implementation is faulty, see comment.

File: cellumata_syntax-v2/cellmata.g4
Comment (falkecarlsen) [reply cellumata_syntax-v2/cellmata.g4]: Agreed. Resolved in 679a9c0

**Review by jenrik — APPROVED** [review]

**Comment by falkecarlsen** [comment]
Good feedback. Yes, we are allowing comments in arbitrary places, this should be fine for now.

**Pull request merged**
